### PR TITLE
Closes #8002 column view handling

### LINKED ISSUE #8002: Column view Handling
**Description**
Make modifications to column view and data transition to RI settings page

**Dependency**
This module is dependent on #7998, #7999 & #8001 to be complete.

**Acceptance Criteria**
- 'See Report' button will be replaced by 'View Details' for completed and successful test
- Clicking 'View Details' will open the RI settings page with the metrics data expanded for the specific test in a new tab
  - This action will be tracked in mixpanel
- For failed, blurred test, hovering on the 'View Details' should display a tooltip explaining why it's deactivated.

### PR DESCRIPTION
**Pull request by Miraeld**
# Description

Fixes #8002 

This PR enhances the Rocket Insights column in post listing pages by replacing the "See Report" button (which linked to GTMetrix) with a "View Details" button that opens the RI settings page with the specific test's metrics expanded. This gives users a more integrated experience to view detailed performance data directly within WP Rocket.

## Type of change

- [x] Enhancement (non-breaking change which improves an existing functionality).

### What was tested

| Scenario | Type | Status |
|----------|------|--------|
| Completed test shows "View Details" button | Manual | ✅ |
| Clicking "View Details" opens RI settings page in new tab | Manual | ✅ |
| URL contains `#rocket_insights={row_id}` for auto-expansion | Manual | ✅ |
| Mixpanel tracking fires on View Details click | Manual | ✅ |
| Blurred test shows disabled "View Details" with tooltip | Manual | ✅ |
| Failed test shows disabled "View Details" with tooltip | Manual | ✅ |
| PHPCS passes on all modified files | Automated | ✅ |
| PHPStan passes on all modified files | Automated | ✅ |

### How to test

**Prerequisites:**
- WP Rocket installed and activated
- Rocket Insights feature enabled
- At least one page tested in Rocket Insights

**Test "View Details" button for completed tests:**
1. Go to Posts > All Posts (or any post type listing)
2. Find a post that has a completed Rocket Insights test (shows a score)
3. Verify the button says "View Details" (not "See Report")
4. Click "View Details"
5. Verify it opens a new tab to WP Rocket settings page
6. Verify the URL ends with `#rocket_insights={test_id}`
7. Verify the test row is automatically expanded (requires PR #8017 to be merged)

**Test disabled state with tooltip (blurred/free user):**
1. Use a free account or simulate blurred state
2. Go to post listing with a tested page
3. Hover over the disabled "View Details" button
4. Verify tooltip shows: "Upgrade to view detailed metrics"

**Test disabled state with tooltip (failed test):**
1. Find a post with a failed Rocket Insights test
2. Hover over the disabled "View Details" button
3. Verify tooltip shows: "Test failed - no details available"

**Test Mixpanel tracking:**
1. Enable Mixpanel opt-in in WP Rocket settings
2. Click "View Details" on a completed test
3. Check browser Network tab for AJAX call to `admin-ajax.php` with action `rocket_track_view_details`

### Affected Features & Quality Assurance Scope

- **Rocket Insights column** in post listing pages (Posts, Pages, CPTs)
- **Mixpanel tracking** for user interactions
- **RI settings page** (integration with auto-expansion from PR #8017)

### Documentation

**Files Modified:**

| File | Purpose |
|------|---------|
| `views/settings/partials/rocket-insights/rocket-insights-column.php` | Template: Changed button label, added tooltips for disabled states |
| `inc/Engine/Admin/RocketInsights/Render.php` | Added `get_view_details_url()` method to generate RI settings URL |
| `inc/Engine/Tracking/Tracking.php` | Added `track_rocket_insights_view_details()` and `ajax_track_view_details()` |
| `inc/Engine/Tracking/Subscriber.php` | Wired up `wp_ajax_rocket_track_view_details` hook |
| `src/js/global/rocket-insights.js` | Added click listeners and AJAX tracking for View Details |
| `src/scss/rocket-insights.scss` | Added `.wpr-ri-view-details-link` styling |

**URL Format:**
```
/wp-admin/options-general.php?page=wprocket#rocket_insights={row_id}
```

The `={row_id}` format is used by PR #8017's auto-expansion feature to automatically open the metrics panel for the specified test.

**Tracking Event:**
- Event name: `Rocket Insights View Details`
- Properties: `context`, `source`, `test_id`

### New dependencies

- **PR #8017** (UI/UX RI settings page) - Required for the auto-expansion feature to work when navigating to the RI settings page.

# Mandatory Checklist

## Code validation

- [x] I validated all the Acceptance Criteria. If possible, provide screenshots or videos.
- [x] I triggered all changed lines of code at least once without new errors/warnings/notices.
- [ ] I implemented built-in tests to cover the new/changed code.

## Code style

- [x] I wrote a self-explanatory code about what it does.
- [x] I protected entry points against unexpected inputs.
- [x] I did not introduce unnecessary complexity.
- [x] Output messages (errors, notices, logs) are explicit enough for users to understand the issue and are actionnable.

## Unticked items justification

- **Built-in tests**: Unit/integration tests for this feature will be added in a follow-up PR once the dependent PRs (#8017, #8009) are merged and the full integration can be tested.

# Additional Checks
- [x] In the case of complex code, I wrote comments to explain it.
- [x] When possible, I prepared ways to observe the implemented system (logs, data, etc.).
- [x] I added error handling logic when using functions that could throw errors (HTTP/API request, filesystem, etc.)

### CODE REVIEW

File: views/settings/partials/rocket-insights/table-row.php [hunk 1]
```diff
@@ -114,3 +118,93 @@
...
+							</div>
+							<div class="metric-value ri-warning">
+								<p>1.4s</p>
+							</div>
+							<div class="metric-value ri-success">
+								<p>700ms</p>
+							</div>
+							<div class="metric-value ri-warning">
+								<p>0.145s</p>
+							</div>
```
[the remark this answers is not in the document]
Comment (jeawhanlee) [reply views/settings/partials/rocket-insights/table-row.php:203]: Will be fixed [here](https://github.com/wp-media/wp-rocket/pull/8021)

File: inc/Engine/Tracking/Tracking.php [hunk 2]
```diff
@@ -282,4 +282,55 @@ public function track_rocket_insights_test( $row_details, $job_details, $plan ):
...
+
+	/**
+	 * Handle AJAX request to track View Details clicks.
+	 *
+	 * @since 3.20.2
+	 *
+	 * @return void
+	 */
+	public function ajax_track_view_details(): void {
+		check_ajax_referer( 'rocket-ajax' );
```
[the remark this answers is not in the document]
Comment (jeawhanlee) [reply inc/Engine/Tracking/Tracking.php]: Fixed

File: src/js/global/ajax.js [hunk 3]
```diff
@@ -671,4 +671,129 @@ document.addEventListener('DOMContentLoaded', function() {
...
+	if ($('.details-section-td').is(':visible')) {
+		var $detailsCells = $('.details-section-td');
+		var $toggleButtons = $('.wpr-ri-item-toggle-single');
+		var $lastToggle = $('.wpr-ri-item-toggle').last();
+		var $lastActions = $('.wpr-ri-item-actions').last();
+		var imgUrl = window.rocket_ajax_data.assets_img_url;
+		var carets = {
+			down: imgUrl + 'ri-caret-down.svg',
+			right: imgUrl + 'ri-caret-right.svg'
+		}
```
[the remark this answers is not in the document]
Comment (jeawhanlee) [reply src/js/global/ajax.js:685]: Not a change in this PR.

File: src/js/global/ajax.js [hunk 4]
```diff
@@ -671,4 +671,129 @@ document.addEventListener('DOMContentLoaded', function() {
...
+				$selectors.lastActions.css('border-bottom-right-radius', 5);
+
+				// Remove border radius for details cell when item is collapsed.
+				$selectors.detailsCells.last().css('border-bottom-left-radius', 0);
+				$selectors.detailsCells.last().css('border-bottom-right-radius', 0);
+
+				// Remove bottom border for details cell when item is collapsed.
+				$selectors.detailsCells.last().css('border-bottom', 0);
+
+				return
```
[the remark this answers is not in the document]
Comment (jeawhanlee) [reply src/js/global/ajax.js:760]: Not a change in this PR.

File: inc/Engine/Admin/RocketInsights/PostListing/Subscriber.php [hunk 5]
```diff
@@ -290,6 +312,36 @@ public function render_rocket_insights_column( string $column, int $post_id ): v
...
+		}
+
+		$row_id  = absint( wp_unslash( $_POST['row_id'] ) );
+		$context = sanitize_text_field( wp_unslash( $_POST['context'] ) );
+
+		if ( ! $this->query->get_row_by_id( $row_id ) ) {
+			wp_send_json_error( 'Invalid row ID' );
+		}
+
+		$this->tracking->track_rocket_insights_view_details( $row_id, $context );
```
Comment (wordpressfan) [inline inc/Engine/Admin/RocketInsights/PostListing/Subscriber.php:340]: Instead of adding tracking as a dependency here, what do u think of taking the following approach to make the tracking more dynamic even for next tracking events:

- Here fire (do_action) a new action called `rocket_send_tracking` and pass two parameters, the first one is the event name (in this case it's `Rocket Insights View Details`) and the second one is the data to be sent (in this case array of row ID and source)
- In tracking subscriber, add one callback to send the mixpanel request based on the passed arguments
- We will use this a lot in the future, so it'd be awesome to have a trait in tracking namespace with a method to fire this new action with those two arguments

**Review by wordpressfan — CHANGES_REQUESTED** [review]
What do u think about this approach?

File: inc/Engine/Admin/RocketInsights/PostListing/Subscriber.php (hunk 5 above)
Comment (jeawhanlee) [reply inc/Engine/Admin/RocketInsights/PostListing/Subscriber.php:340]: @wordpressfan I like the idea, sounds good. Can you please open a new issue for this, so we can work on it in parallel and let this PR move forward.
Thanks 🙏

Comment (wordpressfan) [reply inc/Engine/Admin/RocketInsights/PostListing/Subscriber.php:340]: Sure, created it here:
https://github.com/wp-media/wp-rocket/issues/8047

File: views/settings/partials/rocket-insights/rocket-insights-column.php [hunk 6]
```diff
@@ -136,17 +143,15 @@ class="wpr-ri-test-page <?php echo $data['is_draft'] ? 'wpr-ri-no-credit' : '';
...
-				// See report link - only show if report_url exists.
-				$wpr_report_url = $data['wpr_rocket_row']->report_url ?? ''; // phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals.NonPrefixedVariableFound
-
-				if ( ! empty( $wpr_report_url ) && $data['wpr_can_access_report'] ) :
+				// View details link - only show if test is completed successfully.
+				if ( ! empty( $data['wpr_rocket_row']->id ) && $data['wpr_can_access_report'] ) :
 					?>
-					<a href="<?php echo esc_url( $wpr_report_url ); ?>" class="wpr-ri-see-report-link wpr-icon-report" target="_blank" rel="noopener">
-						<?php esc_html_e( 'See Report', 'rocket' ); ?>
+					<a href="<?php echo esc_url( $data['wpr_view_details_url'] ); ?>" class="wpr-ri-view-details-link wpr-icon-report" target="_blank" rel="noopener" data-rocket-insights-id="<?php echo esc_attr( $data['wpr_rocket_row']->id ); ?>">
```
Comment (wordpressfan) [inline views/settings/partials/rocket-insights/rocket-insights-column.php:149]: Do we still need to open it in a new tab? I think it was like this for the report url so it was external url, I think here we need to open it in the same tab, what do u think?

**Review by wordpressfan — CHANGES_REQUESTED** [review]
One minor suggestion.
Also, I can only see this weird behavior but I'm not sure if it's fixed in another PR or needs to be fixed here:

https://github.com/user-attachments/assets/49cbd4f1-e9c3-472a-9021-4560d7cc15cf

**Review by wordpressfan — APPROVED** [review]
The issue in the screencast above will be fixed here:
https://github.com/wp-media/wp-rocket/issues/8042

So let's move forward with this one not to block the feature.
